### PR TITLE
build: switch to dual package emitting

### PR DESCRIPTION
**Pull request by cf-remylenoir**
# Purpose of PR

> [!NOTE]
> This is only for v5 alpha/beta pre-releases and won't be merged into `main`

Adds setup for emitting 2 package.json files for ESM and CJS, making use of [tsconfig-to-dual-package](https://github.com/azu/tsconfig-to-dual-package).

This approach has the benefit of clearly declaring what type should be used since it relies on 2 package.json and prevents Node/TypeScript configuration confusion or change for the consumers, and mostly issues with our direct dependency for the icons (Phosphor)

## Motivation and readings 🍿 
- https://github.com/microsoft/TypeScript/issues/49462
- https://github.com/microsoft/TypeScript/issues/54593

### CODE REVIEW

File: .changeset/config.json
Comment (cf-remylenoir) [inline .changeset/config.json:1]: Remnant of publishing the wrong package during some testing. To be removed and published package to be unpublished.

File: packages/components/icon/package.json [hunk 1]
```diff
@@ -1,29 +1,35 @@
...
     },
     "./package.json": "./package.json"
   },
   "source": "src/index.ts",
-  "types": "dist/types/index.d.ts",
   "sideEffects": false,
   "type": "module",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/esm/index.d.ts",
```
Comment (cf-remylenoir) [inline packages/components/icon/package.json:25]: For backward compability

File: .changeset/config.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     }
   ],
   "ignore": [
+    "@contentful/f36-icon-alpha",
```
Comment (cf-remylenoir) [inline .changeset/config.json:10]: Remnant of publishing the wrong package during some testing. To be removed and [published package](https://www.npmjs.com/package/@contentful/f36-icon-alpha) to be unpublished.

File: packages/components/icon/package.json [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
+      },
+      "default": "./dist/cjs/index.js"
     },
     "./package.json": "./package.json"
   },
-  "source": "src/index.ts",
-  "types": "dist/types/index.d.ts",
   "sideEffects": false,
   "type": "module",
+  "main": "./dist/cjs/index.js",
```
Comment (andipaetzold) [inline packages/components/icon/package.json:22]: `type: module` & `main` being CJS seems wrong

File: packages/components/icon/package.json [hunk 4]
```diff
@@ -49,5 +53,8 @@
   "publishConfig": {
     "registry": "https://npm.pkg.github.com/",
     "access": "public"
+  },
+  "engines": {
+    "node": ">=16"
```
Comment (andipaetzold) [inline packages/components/icon/package.json:58]: Why is this required? It's a frontend package.

Comment (cf-remylenoir) [reply packages/components/icon/package.json:58]: It isn't, to be removed.
